akka-projection-slick/src/main/scala/akka/projection/slick/SlickProjection.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.slick
+
+import akka.Done
+import akka.actor.ClassicActorSystemProvider
+import akka.annotation.{ ApiMayChange, InternalApi }
+import akka.projection.Projection
+import akka.stream.scaladsl.{ Keep, Sink, Source }
+import akka.stream.{ KillSwitch, KillSwitches }
+import org.slf4j.LoggerFactory
+import slick.basic.DatabaseConfig
+import slick.dbio.DBIO
+import slick.jdbc.JdbcProfile
+
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+
+object SlickProjection {
+  @ApiMayChange
+  def transactional[Offset, StreamElement, P <: JdbcProfile: ClassTag](
+      projectionId: String,
+      sourceProvider: Option[Offset] => Source[StreamElement, _],
+      offsetExtractor: StreamElement => Offset,
+      databaseConfig: DatabaseConfig[P])(eventHandler: StreamElement => DBIO[Done]): Projection =
+    new SlickProjectionImpl(projectionId, sourceProvider, offsetExtractor, databaseConfig, eventHandler)
```

Comment:
I'm not so convinced of this API. It seems that there are a lot that a user needs to assemble. On the other hand, this is not day-to-day code. You usually write it a few times for a given project.